mondrian/src/it/java/mondrian/rolap/NonEmptyTest.java

```diff
@@ -5186,7 +5187,7 @@ public void testMondrian1133() {
             + "    `store`.`store_country`,\n"
             + "    `store`.`store_state`\n"
             + "having\n"
-            + "    UPPER(c1) REGEXP '.*CA.*'\n"
+            + "    UPPER(c1) REGEXP '.*CA.*' AND UPPER(c1) IS NOT NULL\n"
```

Comment:
It would be more performant to have this read:
c1 IS NOT NULL AND UPPER(c1) REGEXP '.*CA.*'\n

1- No need to do upper(NULL) - checking null first makes more sense
2- You eliminate the nulls first so that the REGEXP isn't evaluated for the null rows